lib/src/analyzers/lint_analyzer/rules/rules_list/use_setstate_synchronously/visitor.dart

```diff
@@ -65,7 +72,6 @@ class _AsyncSetStateVisitor extends RecursiveAstVisitor<void> {
       return node.visitChildren(this);
     }
 
-    node.condition.visitChildren(this);
```

Comment:
@Desdaemon removing this looks safe or am I missing something?

Reply:
This might cause lints to fail for e.g. `if (await ..)`, but I was mostly being cautious when I wrote this.

Reply:
Hm, I'll add some tests to check this case

Reply:
Actually, it was not checked coz of `.visitChildren`, I've added a test and fixed the problem